src/traces/contour/hover.js

```diff
@@ -9,8 +9,26 @@
 
 'use strict';
 
+var Color = require('../../components/color');
+
 var heatmapHoverPoints = require('../heatmap/hover');
 
 module.exports = function hoverPoints(pointData, xval, yval, hovermode, hoverLayer) {
-    return heatmapHoverPoints(pointData, xval, yval, hovermode, hoverLayer, true);
+    var hoverData = heatmapHoverPoints(pointData, xval, yval, hovermode, hoverLayer, true);
+
+    if(hoverData) {
+        hoverData.forEach(function(hoverPt) {
+            var trace = hoverPt.trace;
+            if(trace.contours.type === 'constraint') {
+                if(trace.fillcolor && Color.opacity(trace.fillcolor)) {
+                    hoverPt.color = Color.addOpacity(trace.fillcolor, 1);
+                }
+                else if(trace.contours.showlines && Color.opacity(trace.line.color)) {
+                    hoverPt.color = Color.addOpacity(trace.line.color, 1);
+                }
+            }
+        });
+    }
+
+    return hoverData;
```

Comment:
Moverover, should we show hover labels for x/y/z points outside the constraints? I'm not sure what's best here. Maybe that's why Ricky didn't bother implementing hover labels for `contourcarpet` in the first place.

Reply:
One more comment: looks like contours would benefit from a version of `hovermode: 'compare'` that allows showing multiple labels of super-imposed traces as proposed in https://github.com/plotly/plotly.js/issues/720#issuecomment-230900670

Currently, `hovermode: 'closest'` isn't that useful on the `contour_constraints` mock:

![peek 2018-01-29 18-35](https://user-images.githubusercontent.com/6675409/35540370-968799b8-0523-11e8-8a9d-2fdd2242555b.gif)

Reply:
> Moverover, should we show hover labels for x/y/z points outside the constraints?

I think so. In fact it's possible that there actually *aren't* any data points (only interpolations) within the constraints, for `[]` type. Anyway I could see people using hover to see "how bad would it be to operate out here"

> One more comment: looks like contours would benefit from a version of `hovermode: 'compare'` that allows showing multiple labels of super-imposed traces as proposed in https://github.com/plotly/plotly.js/issues/720#issuecomment-230900670

Yes definitely - but perhaps we tackle that all at once as part of #720? There are going to be some weird cases with contours, like if two constraints overlay but with misaligned grids.

Reply:
> Yes definitely - but perhaps we tackle that all at once as part of #720?

Absolutely :ok_hand: